File: tyama_icpc2007rA.hs
#!/usr/bin/env runghc
import Control.Applicative
import Data.List

calc n k = foldl' (\r i -> mod (r+k) i) 0 [1..n-1]

main = map read . words <$> getLine >>= \[n,k,m] ->
	if n/=0 then do
		print $ ((mod ((calc n k)+m) n)+1)
		main
	else return ()

File: tyama_icpc2007rA.erl
#!/usr/bin/escript

-module(prog).
-export([main/0]).

calc(R,I,N,K) ->
	if
		I==N -> R;
		I/=N -> calc((R+K) rem I,I+1,N,K)
	end.

main() ->
	X=io:get_line(standard_io,""),
	A=array:from_list(string:tokens(string:strip(X,right,10)," ")),
	N=list_to_integer(array:get(0,A)),
	K=list_to_integer(array:get(1,A)),
	M=list_to_integer(array:get(2,A)),
	if
		N==0 -> true;
		N/=0 -> io:format("~w\n",[(calc(0,1,N,K)+M) rem N+1]),main()
	end.